hatch/src/deps/dependency.rs

```diff
@@ -11,15 +11,9 @@ pub struct Dependency {
 impl Dependency {
   pub fn new(url: String) -> Dependency {
     Dependency {
-      name: PathBuf::from(OsString::from(url.clone()))
-        .components()
-        .last()
-        .unwrap()
-        .as_os_str()
-        .to_string_lossy()
-        .into_owned()
-        .as_str()
-        .replace(".git", ""),
+      name: (|u: &OsStr| -> String {
```

Comment:
This is pretty dense :(. Maybe make this whole thing a local function or something?  Maybe make a variable for the PathBuf?

Reply:
fineee